### PR TITLE
Rustin patch having

### PR DESCRIPTION
**Pull request by weiznich**
Workaround Github Bug which prevents reopening #2740
cc @hi-rustin 

close ​#2581​
close ​#2590​
close ​#2745​
This PR is based on ​#2590,​ thanks dnamsons for the great work.

### CODE REVIEW

**Review by pksunkara — APPROVED** [review]

File: diesel_compile_tests/tests/fail/having_cant_be_used_without_group_by.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+allow_tables_to_appear_in_same_query!(users, posts);
+
+fn main() {
+    let conn = PgConnection::establish("").unwrap();
+
+    users::table.select(users::name).having(users::id.gt(1)).load(&conn);
+
+    users::table.into_boxed().having(users::id.gt(1)).load(&conn);
+
+    users::table.select(users::name).group_by(users::id).having(posts::id.eq(42)).load(&conn);
```
Comment (weiznich) [inline diesel_compile_tests/tests/fail/having_cant_be_used_without_group_by.rs:30]: Can we have this test also for boxed select statements?

```suggestion
    users::table.select(users::name).group_by(users::id).having(posts::id.eq(42)).load(&conn);

    users::table.select(users::name).group_by(users::id).into_boxed().having(posts::id.eq(42)).load(&conn);
```

**Review by weiznich — COMMENTED** [review]
This one tests should be added, otherwise this is fine :+1: @hi-rustin Thanks for working on this.

File: diesel_compile_tests/tests/fail/having_cant_be_used_without_group_by.rs (hunk 1 above)
Comment (hi-rustin) [reply diesel_compile_tests/tests/fail/having_cant_be_used_without_group_by.rs:30]: Added, thanks to all of you for your patient review and help!

**Comment by hi-rustin** [comment]
> Workaround Github Bug which prevents reopening #2740
> cc @hi-rustin

Maybe we should change the body to:

```md
close #2581
close #2590
close #2745
This PR is based on #2590, thanks dnamsons for the great work.
```

**Pull request merged**